CI: TestPrivilegedSocketDestroyers: `creating map: pin map to /sys/fs/bpf/tc/globals/cilium_lb6_reverse_sk: no such file or directory`

<details><summary>Test output</summary>

```
--- FAIL: TestPrivilegedSocketDestroyers (0.03s)

    sockets_test.go:319:
            Error Trace:    /host/pkg/datapath/sockets/sockets_test.go:252
                            /host/pkg/datapath/sockets/sockets_test.go:319
                            /host/pkg/datapath/sockets/sockets_test.go:421
            Error:          Received unexpected error:
                            creating map: pin map to /sys/fs/bpf/tc/globals/cilium_lb6_reverse_sk: no such file or directory
            Test:           TestPrivilegedSocketDestroyers

FAIL    github.com/cilium/cilium/pkg/datapath/sockets   0.160s
```

</details>

https://github.com/cilium/cilium/actions/runs/23456655634/job/68247821634

The bpffs path `/sys/fs/bpf/tc/globals/` does not exist when the test
tries to pin the `cilium_lb6_reverse_sk` map. Likely a race with bpffs
setup in the CI environment.

cc @cilium/sig-datapath

--- a/pkg/datapath/sockets/sockets_test.go
+++ b/pkg/datapath/sockets/sockets_test.go
@@ -418,6 +418,8 @@ func TestPrivilegedSocketDestroyers(t *testing.T) {
 	testutils.PrivilegedTest(t)
 	log := hivetest.Logger(t)
 
+	bpf.CheckOrMountFS(log, "")
+
 	socketDestroyers := makeSocketDestroyers(t)
 	servers := map[string][]string{
 		"127.0.0.1:8888": {"udp", "tcp"},

